### PR TITLE
Support for liveness probe for ovs-cni marker

### PR DESCRIPTION
**Pull request by rthakur-est**
This change adds support for file based liveness probe for ovs-cni marker.
This is done by changing the modification time of "/tmp/healthy" file
from the marker code. And the exec probe verifies that the file has been
modified in last one minute.

#230 

Signed-off-by: Ravindra Thakur <ravindra.nath.thakur@est.tech>

### CODE REVIEW

File: cmd/marker/main.go [hunk 1]
```diff
@@ -108,3 +113,24 @@ func main() {
 
 	}, time.Duration(*updateInterval)*time.Second, 1.2, true, wait.NeverStop)
 }
+
+func keepAlive() {
```
Comment (phoracek) [inline cmd/marker/main.go:117]: Would you please introduce a new test into https://github.com/k8snetworkplumbingwg/ovs-cni/blob/main/tests/marker_test.go verifying that the pod survives the health check and does not get restarted?

**Review by phoracek — CHANGES_REQUESTED** [review]
Looks and works well, thanks for the contribution. I left some small comments.

File: manifests/ovs-cni.yml.in [hunk 2]
```diff
@@ -69,6 +69,15 @@ spec:
...
                 fieldPath: spec.nodeName
+        livenessProbe:
+          exec:
+            command:
+              - sh
+              - -c
+              - >-
+                find /tmp/healthy -mmin -1 | grep -q '/tmp/healthy'
+          initialDelaySeconds: 15
+          periodSeconds: 15
```
[the remark this answers is not in the document]
Comment (rthakur-est) [reply manifests/ovs-cni.yml.in:80]: I completely agree with this. I will change the timer intervals to 1 minute.

**Review by phoracek — APPROVED** [review]
/lgtm
/approve

Fantastic, thanks a lot @rthakur-est ! I failed the check offline and it restarted the pod as expected.

**Comment by phoracek** [comment]
/retest

Failed on timeout. I will post a patch increasing the 10 minute test timeout

**Comment by rthakur-est** [comment]
@phoracek Thanks for the quick review.

**Pull request merged**